platform/openide.util.ui/test/unit/src/org/openide/xml/XMLUtilTest2.java

```diff
@@ -364,7 +364,11 @@ public void testSignificantWhitespace() throws Exception {
         String data =
                 "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" +
                 "<r>\n" +
-                "    <p>This is <em>not</em> a test!</p>\n" +
```

Comment:
i remember looking at this before and I couldn't figure out if the test was supposed to assert that NB isn't re-formatting the code or if is about something else

Reply:
F*** we might indeed see a Xalan bug here. Not sure what to make of this.

Reply:
i have a few attempts on those tests in my stash. I remember trying to influence how the whitespace is handled. Let me finish the maven PR I am working on first, then I take a look at the stash.

Reply:
Would be interesting. At this point I'm inclined to say, that the default ident feature of xalan does not preserve significant white space and the solution might be to not let the xlst processor do the indention, but do it ourselves.

Reply:
I found what triggered my change. From the documentation of `XMLUtil#write`:

> The precise output format is not guaranteed but this method will attempt to indent it sensibly.

My fix is still broken, but ....

Reply:
so i looked at this probably 2 years ago, but I remembered a few things after looking at the stash:
 - JDK 9-13 handles whitespace in CDATA differently, `testCDATA()` test fails only in that range -> my suggestion: lets forget about JDK 11 and go right to 17+
 - the `testSignificantWhitespace()` fails from 9+
 - `XMLUtil.parse` / `XMLUtil.write` **is always** reformatting the document. The difference is that on 8, a tag+text sequence does not cause a newline, but on 17+ it does. I suppose this is what "Significant Whitespace" means? Never heard of it before but i got a few hits - will take another look some time later.

Reply:
https://bugs.openjdk.org/browse/JDK-8223291 this is likely the CDATA issue, it matches the 9-13 JDK range

Reply:
I am to 90% sure that there is no real solution to this. Since the method would have to detect mixed content and switch from xml to html mode to be able to make the test happy. The doc on top of the method doesn't really specify how it formats anything.

Since NB is working fine on JDK 9+ this implementation detail of mixed content formatting doesn't seem to be a problem.

Lets turn `testSignificantWhitespace()` off, and get this in :)